src/Coriolis/general_f_plane.jl

```diff
@@ -0,0 +1,63 @@
+using Oceananigans.Grids: ZDirection, validate_unit_vector
+
+"""
+    GeneralFPlane{FT} <: AbstractRotation
+
+A Coriolis implementation that accounts for the locally vertical and
+both local horizontal components of the rotation vector. Traditionally
+(see [`FPlane`](@ref)) only the locally vertical component is accounted for.
+"""
+struct GeneralFPlane{FT} <: AbstractRotation
+    fx :: FT
+    fy :: FT
+    fz :: FT
+end
+
+"""
+    GeneralFPlane([FT=Float64;] coriolis_frequency=2Ω_Earth, rotation_axis=ZDirection(), latitude=nothing)
+
+Returns a parameter object for constant rotation about an axis `rotation_axis` with
+`coriolis_frequency` decomposed into the `x`, `y` and `z` directions accordingly. In oceanography
+the components `x`, `y`, `z` correspond to the directions east, north, and up.
+
+`latitude` has to be in degrees (not radians!) and if it is specified, the argument for
+`rotation_axis` gets overwritten and the rotation axis is calculated as `[0, cosd(latitude),
+sind(latitude)]`. By default, `coriolis_frequency` is assumed to be the Coriolis frequency at
+Earth's north pole (1.458423e-4 1/s).
+"""
+function GeneralFPlane(FT=Float64; coriolis_frequency=2Ω_Earth, rotation_axis=ZDirection(), latitude=nothing)
+    if !isnothing(latitude)
+        rotation_axis = [0, cosd(latitude), sind(latitude)]
+    end
+    rotation_axis = validate_unit_vector(rotation_axis)
+
+    if rotation_axis isa ZDirection
+        fx = fy = 0
+        fz = coriolis_frequency
+    else
+        fx = coriolis_frequency * rotation_axis[1]
+        fy = coriolis_frequency * rotation_axis[2]
+        fz = coriolis_frequency * rotation_axis[3]
+    end
+
+    return GeneralFPlane{FT}(fx, fy, fz)
+end
+
+
+
+# This function is eventually interpolated to fcc to contribute to x_f_cross_U.
+@inline fʸw_minus_fᶻv(i, j, k, grid, coriolis, U) =
+    coriolis.fy * ℑzᵃᵃᶜ(i, j, k, grid, U.w) - coriolis.fz * ℑyᵃᶜᵃ(i, j, k, grid, U.v)
+
+@inline fᶻu_minus_fˣw(i, j, k, grid, coriolis, U) =
```

Comment:
I have never seen the `fx` compoments before and nice to see them for the first time!

Reply:
Hopefully I did the math right :joy: 

But that's also part of the reason why I think we could test that Coriolis is doing what it's supposed to be doing

Reply:
I double checked and the equations look good to me.